# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [22]:
# Выведем данные по количеству детей среди тех, кто вернул кредит в срок: 
debt_0 = data.loc[data['debt'] == 0]
debt_0['children'].value_counts()

0    13028
1     4364
2     1858
3      303
4       37
5        9
Name: children, dtype: int64

In [23]:
# Выведем данные по количеству детей среди всех кредитовзятелей:
data['children'].value_counts()

0    14091
1     4808
2     2052
3      330
4       41
5        9
Name: children, dtype: int64

In [24]:
# Создадит таблицу, с данными из двух предыдущих шагов:
children = [
    ['0', 14091, 13028],
    ['1', 4808, 4364],
    ['2', 2052, 1858],
    ['3', 330, 303],
    ['4', 41, 37],
    ['5', 9, 9]
]
columns = ['children', 'all', 'close_without debt']
data_children = pd.DataFrame(data = children, columns = columns)

# Добавим в таблицу дополнительный столбец '%'с данными о том, какой процент кредитовзятелей из общего числа закрыли кредит в срок: 
data_children['%'] = data_children['close_without debt'] / data_children['all']
data_children.sort_values(by='%', ascending = False)

children    all  close_without debt         %
5        5      9                   9  1.000000
0        0  14091               13028  0.924562
3        3    330                 303  0.918182
1        1   4808                4364  0.907654
2        2   2052                1858  0.905458
4        4     41                  37  0.902439

**Вывод:**
Заемщики с 5-ю детьми имеют 100% закрываемость кредитов. Далее за ними идут заемщики без детей, у них процент закрытия равен 92%. Наихудщая закрываеомсть кредитов у заемщиков с 4 детьми (90%)

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [25]:
# Выведем данные по семейному положению среди тех, кто вернул кредит в срок: 
debt_0['family_status'].value_counts()

женат / замужем          11334
гражданский брак          3749
Не женат / не замужем     2523
в разводе                 1105
вдовец / вдова             888
Name: family_status, dtype: int64

In [26]:
# Выведем данные по семейному положению среди всех кредитовзятелей:
data['family_status'].value_counts()

женат / замужем          12261
гражданский брак          4134
Не женат / не замужем     2796
в разводе                 1189
вдовец / вдова             951
Name: family_status, dtype: int64

In [27]:
# Создадит таблицу, с данными из двух предыдущих шагов:
family_status = [
    ['женат / замужем', 12261, 11334],
    ['гражданский брак', 4134, 3749],
    ['Не женат / не замужем', 2796, 2523],
    ['в разводе', 1189, 1105],
    ['вдовец / вдова', 951, 888]
]
columns = ['family_status', 'all', 'close_without debt']
data_family_status = pd.DataFrame(data = family_status, columns = columns)

# Добавим в таблицу дополнительный столбец '%'с данными о том, какой процент кредитовзятелей из общего числа закрыли кредит в срок: 
data_family_status['%'] = data_family_status['close_without debt'] / data_family_status['all'] 
data_family_status.sort_values(by='%', ascending = False)

family_status    all  close_without debt         %
4         вдовец / вдова    951                 888  0.933754
3              в разводе   1189                1105  0.929352
0        женат / замужем  12261               11334  0.924394
1       гражданский брак   4134                3749  0.906870
2  Не женат / не замужем   2796                2523  0.902361

**Вывод:** 
Заемщики в семейном положении "вдовец/вдова" имеют наилучший % закрытия кредита в срок (93%). Наихудший % закрытия у тех, кто в гражданском браке (90%).

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [28]:
# Выведем данные по уровню дохода среди тех, кто вернул кредит в срок: 
debt_0['total_income_category'].value_counts()

C    14568
B     4660
D      328
A       23
E       20
Name: total_income_category, dtype: int64

In [29]:
# Выведем данные по семейному положению среди всех кредитовзятелей:
data['total_income_category'].value_counts()

C    15921
B     5014
D      349
A       25
E       22
Name: total_income_category, dtype: int64

In [30]:
# Создадит таблицу с данными из двух предыдущих шагов:
total_income_category = [
    ['50 001 - 200 000', 15921, 14568],
    ['200 001 - 1 000 000', 5014, 4660],
    ['30 001 - 50 000', 349, 328],
    ['1 000 001 и выше', 25, 23],
    ['0 - 30 000', 22, 20]
]
columns = ['total_income_category', 'all', 'close_without debt']
data_total_income_category = pd.DataFrame(data = total_income_category, columns = columns)

# Добавим в таблицу дополнительный столбец '%'с данными о том, какой процент кредитовзятелей из общего числа закрыли кредит в срок: 
data_total_income_category['%'] = data_total_income_category['close_without debt'] / data_total_income_category['all'] 
data_total_income_category.sort_values(by='%', ascending = False)

total_income_category    all  close_without debt         %
2       30 001 - 50 000    349                 328  0.939828
1   200 001 - 1 000 000   5014                4660  0.929398
3      1 000 001 и выше     25                  23  0.920000
0      50 001 - 200 000  15921               14568  0.915018
4            0 - 30 000     22                  20  0.909091

**Вывод:** 
Заемщики c доходом от "30 001 до 50 000" имеют наилучший % закрытия кредита в срок (93,98%). Наихудший % закрытия у заемщиков с доходом до 30 000.

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [31]:
# Выведем данные по целям среди тех, кто вернул кредит в срок:
debt_0['purpose_category'].value_counts()

операции с недвижимостью    9971
операции с автомобилем      3879
получение образования       3619
проведение свадьбы          2130
Name: purpose_category, dtype: int64

In [32]:
# Выведем данные по целям среди всех кредитовзятелей:
data['purpose_category'].value_counts()

операции с недвижимостью    10751
операции с автомобилем       4279
получение образования        3988
проведение свадьбы           2313
Name: purpose_category, dtype: int64

In [33]:
# Создадит таблицу с данными из двух предыдущих шагов:
purpose = [
    ['операции с недвижимостью', 10751, 9971],
    ['операции с автомобилем', 4279, 3879],
    ['получение образования', 3988, 3619],
    ['проведение свадьбы', 2313, 2130]
]
columns = ['purpose', 'all', 'close_without debt']
data_purpose = pd.DataFrame(data = purpose, columns = columns)

# Добавим в таблицу дополнительный столбец '%'с данными о том, какой процент кредитовзятелей из общего числа закрыли кредит в срок: 
data_purpose['%'] = data_purpose['close_without debt'] / data_purpose['all'] 
data_purpose.sort_values(by='%', ascending = False)

purpose    all  close_without debt         %
0  операции с недвижимостью  10751                9971  0.927449
3        проведение свадьбы   2313                2130  0.920882
2     получение образования   3988                3619  0.907472
1    операции с автомобилем   4279                3879  0.906520

**Вывод:** 
У кредитов с целью "операции с недвижимостью" наилучшее закрывание (92,7%). Далее идут кредиты с целью "проведение свадьбы" (92%). Наихудшая закрываемость у кредитов с целью "на операции с автомобилем" (90%). 

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

In [34]:
# повторно импортируем таблицу и выведем только те столбцы, где одновременно пропуски значений в 'days_employed' и 'total_income' 
import pandas as pd
data_nan = pd.read_csv('/datasets/data.csv')
data_nan.loc[(data_nan['days_employed'].isna()) & (data_nan['total_income'].isna())].head(30)

children  days_employed  dob_years            education  education_id  \
12          0            NaN         65              среднее             1   
26          0            NaN         41              среднее             1   
29          0            NaN         63              среднее             1   
41          0            NaN         50              среднее             1   
55          0            NaN         54              среднее             1   
65          0            NaN         21              среднее             1   
67          0            NaN         52               высшее             0   
72          1            NaN         32               высшее             0   
82          2            NaN         50               высшее             0   
83          0            NaN         52              среднее             1   
90          2            NaN         35               высшее             0   
94          1            NaN         34               высшее             0   
96          0            NaN         44              СРЕДНЕЕ             1   
97          0            NaN         47               высшее             0   
120         0            NaN         46              среднее             1   
121         0            NaN         29               высшее             0   
135         0            NaN         27              среднее             1   
141         0            NaN         39              среднее             1   
145         0            NaN         62              среднее             1   
174         0            NaN         55               высшее             0   
181         0            NaN         26              среднее             1   
189         1            NaN         30              среднее             1   
205         1            NaN         31               высшее             0   
220         1            NaN         23  неоконченное высшее             2   
241         0            NaN         47              среднее             1   
242         0            NaN         58              среднее             1   
247         1            NaN         60               высшее             0   
250         0            NaN         54               высшее             0   
264         2            NaN         40              среднее             1   
278         1            NaN         23              Среднее             1   

             family_status  family_status_id gender  income_type  debt  \
12        гражданский брак                 1      M    пенсионер     0   
26         женат / замужем                 0      M  госслужащий     0   
29   Не женат / не замужем                 4      F    пенсионер     0   
41         женат / замужем                 0      F  госслужащий     0   
55        гражданский брак                 1      F    пенсионер     1   
65   Не женат / не замужем                 4      M    компаньон     0   
67         женат / замужем                 0      F    пенсионер     0   
72         женат / замужем                 0      M  госслужащий     0   
82         женат / замужем                 0      F    сотрудник     0   
83         женат / замужем                 0      M    сотрудник     0   
90         женат / замужем                 0      F    сотрудник     0   
94        гражданский брак                 1      F    компаньон     0   
96         женат / замужем                 0      F    сотрудник     0   
97         женат / замужем                 0      F    сотрудник     0   
120        женат / замужем                 0      F    сотрудник     0   
121        женат / замужем                 0      F    компаньон     0   
135        женат / замужем                 0      M    компаньон     0   
141       гражданский брак                 1      M    сотрудник     0   
145        женат / замужем                 0      M    пенсионер     0   
174         вдовец / вдова                 2      F    компаньон     0   
181       гражданский брак      

Анализ значений NaN для столбцов 'days_employed' (общий трудовой стаж в днях) и 'total_income' (ежемесячный доход) не выявил каких-либо закономерностей с другими столбцами (были гипотезы, что NaN может совпадать с, например, определенным типом занятости - например, быть присущей только пенсионерам или только тем, кто является компаньоном. Такие случаи не были выявлены. 

Так как у данных столбцов одинаковое количество NaN (= 2174) можно предположить, что отсутствие значений в данных столбцах может быть присуще какому-то одному региону/банку/сотруднику. Возможно, сотрудники данного региона/банка не знают, что данные поля обязательны для заполненения (либо у них нет соответствующей настройки на ПК).  

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

Заполнить пропуски медианным значение лучшее решение для количественных переменных потому что среднее арифметическое или медиана - это значения характеризующие состояние выборки. Данные значение не исказят состояние значений для выборки. 

### Шаг 4: общий вывод.

Рекомендую: 
- выяснить истинные причины отсутствия значений для столбцов 'общий трудовой стаж в днях' и 'ежемесячный доход'. Вероятная причина - незаполнение столбцов присуще определенному/ным регуону/банку/сотруднику;
- ввести для столбца 'цель получения кредита' категоризатор на уровне переноса анкеты в базу данных (то есть при вводе должен быть ограниченный список целей).

Общие выводы по исследованию. Рекомендую использовать результаты в скоринговых моделях: 
- заемщики с 5-ю детьми имеют 100% закрываемость кредитов. Далее за ними идут заемщики без детей - у них процент закрытия равен 92%. Наихудщая закрываеомсть кредитов у заемщиков с 4-мя детьми (90%);

- заемщики в семейном положении "вдовец/вдова" имеют наилучший % закрытия кредита в срок (93%). Наихудший % закрытия у тех, кто в гражданском браке (90%);

- заемщики c доходом от "30 001 до 50 000" имеют наилучший % закрытия кредита в срок (93,98%). Наихудший % закрытия у заемщиков с доходом до 30 000;

- у кредитов с целью "операции с недвижимостью" наилучшее закрывание (92,7%). Далее идут кредиты с целью "проведение свадьбы" (92%). Наихудшая закрываемость у кредитов с целью "на операции с автомобилем" (90%). 


### Подготовительный этап
В ходе исследования нами были обработаны данные, полученные из файла 'data.csv'. Данные содержат статистику о платёжеспособности клиента.
    
В ходе исследования для кредитного отдела банка на предварительном этапе были проведены следующие работы: 
- обнаружены пропуски значений в столбцах 'days empliyed' и 'total_income';
- для корректности обработки данных, строки с пропусками в 'days empliyed' и 'total_income' были заменены медианными значениями присущими для соответствующей выборки. Данный тип заполнения пропушенных данных позволяет получить большую выборку для анализы (т.к. мы не удаляем значения), но и не исказить результаты значениями, не присущими для данной выборки;
- в столбце 'days_employed' и 'children' были обнаружены анамальные значения. Данные значения были исправлены;
- столбцы 'education' и 'total_income' были приведенены к значениям, доступным для обработки;
- строки-дубликаты были удаленены из всех таблицы (их было порядка 71);
- в таблицу был добавлен новый столбец 'total_income_category' - всем заемщикам банка была присвоена категория, соответствующая их уровню дохода. Так, были определенены категории: 
    - до 30 000 — категория 'E';
    - от 30 001 – до 50 000 — категория 'D';
    - от 50 001 – до 200 000 — категория 'C';
    - от 200 001 – до  1 000 000 — категория 'B';
    - от 1 000 001 и выше — категория 'A'.
- в таблице с данными было заявлено порядка 38 целей кредита. Мы тщательно изучили эти цели и выделили 4 типа целей: 
    - операции с автомобилем;
    - операции с недвижимостью;
    - проведение свадьбы;
    - получение образования.

    В таблицу был добавлен новый столбец 'purpose' - всем заемщикам банка была определенена цель исходя из заявленного выше классификатора. 

### Выводы по целям исследования
Перед нами была поставлена цель разобраться, влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок. 
Нами получены следующие результаты.
    
#### Влияние семейного положения
Семейное положение влияет на закрытие кредита в срок. 

**Cемейное** **положение**  |  **%** **закрытий** **без** **просрочки**
 ***
    - вдовец/вдова          |   93,3%
    - в разводе             |   92,9%
    - женат/замужен         |   92,4%
    - гражданский брак      |   90,7%
    - не женат / не замужем |   90,2%

Заемщики состоящие или состоявшие в браке ранее - реже допускают просрочку. 

#### Влияние количества детей у займщика
Количество детей влияет на наличие просрочки.

**Количество** **детей**    |  **%** **закрытия** **без** **просрочки**
***
    - 5   |   100%
    - 0   |   92,5%
    - 3   |   91,8%
    - 1   |   90,8%
    - 2   |   90,5%
    - 4   |   90,1%

У заемщиков с детьми больше вероятность просрочки (в категории 5 и более детей всего 9 заемщиков, выборку по ним нельзя считать репрезентативной). 
 
### Дополнительные выводы
Нами были проведенены дополнительные исследования данных заемщиков банка. Рекомендуем также использовать их при построении модели кредитного скоринга.  

#### Влияние уровня дохода
Уровень дохода влияет на возврат кредита без просрочек. 

**Уровень** **дохода**     |  **%** **закрытия** **без** **просрочки**
***    
    - от 50 001 - до 200 000     |   94%
    - от 200 001 - до 1 000 000  |   93%
    - от 30 001 - до 50 000      |   92%
    - от 1 000 001 и выше        |   91,5%
    - до 30 000                  |   90,9%

Более состоятельные клиенты - более надежные заемщики. 

#### Влияние цели кредита
Цель кредита влияет на возврат кредита без просрочек.

**Цель** **кредита**            |  **%** **закрытия** **без** **просрочки**
***
    - операции с недвижимостью  |   92,7% 
    - операции с автомобилем    |   92,1%
    - получение образования     |   90,7%
    - проведение свадьбы        |   90,7%

Ипотечники являются самой добросовестной категорией заемщиков.
    
### Рекомендации по работе с данными
    
- мы рекомендуем выяснить истинные причины отсутствия значений для столбцов 'общий трудовой стаж в днях' и 'ежемесячный доход'. Вероятная причина - незаполнение столбцов присуще определенному/ным регуону/банку/сотруднику. Также вероятны технические ошибки. Больший объем данных позволяет делать более точные прогнозы;
- мы рекомендуем ввести для столбца 'цель получения кредита' категоризатор на уровне переноса анкеты в базу данных (то есть при вводе должен быть ограниченный список целей).

</div>

